graylog2-web-interface/src/components/configurations/PermissionsConfig.tsx

```diff
@@ -0,0 +1,138 @@
+import { PermissionsConfigType } from 'src/stores/configurations/ConfigurationsStore';
+
+import { Button, Col, Modal, Row } from 'components/bootstrap';
+import FormikInput from 'components/common/FormikInput';
+import Spinner from 'components/common/Spinner';
+import { InputDescription } from 'components/common';
+
+type Props = {
+  config: PermissionsConfigType,
+  updateConfig: (config: PermissionsConfigType) => Promise<void>,
+};
+
+const StyledDefList = styled.dl.attrs({
+  className: 'deflist',
+})(({ theme }: { theme: DefaultTheme }) => css`
+  &&.deflist {
+    dd {
+      padding-left: ${theme.spacings.md};
+      margin-left: 200px;
+    }
+  }
+`);
+
+const LabelSpan = styled.span(({ theme }: { theme: DefaultTheme }) => css`
+  margin-left: ${theme.spacings.sm};
+  font-weight: bold;
+`);
+
+const PermissionsConfig = ({ config, updateConfig }: Props) => {
+  const [showModal, setShowModal] = useState<boolean>(false);
+
+  const _saveConfig = (values) => {
+    updateConfig(values).then(() => {
+      setShowModal(false);
+    });
+  };
+
+  const _resetConfig = () => {
+    setShowModal(false);
+  };
+
+  return (
+    <div>
+      <h2>Permissions Config</h2>
+      <p>These settings can be used to control which entity sharing options are available</p>
+
+      {!config ? <Spinner /> : (
+        <>
+          <StyledDefList>
+            <dt>Share with everyone:</dt>
+            <dd>{config.allow_sharing_with_everyone ? 'Enabled' : 'Disabled'}</dd>
+            <dt>Share with users:</dt>
+            <dd>{config.allow_sharing_with_users ? 'Enabled' : 'Disabled'}</dd>
+          </StyledDefList>
+
+          <p>
+            <Button type="button"
+                    bsSize="xs"
+                    bsStyle="info"
+                    onClick={() => {
+                      setShowModal(true);
+                    }}>Configure
+            </Button>
+          </p>
+
+          <Modal show={showModal} onHide={_resetConfig} aria-modal="true" aria-labelledby="dialog_label">
+            <Formik onSubmit={_saveConfig} initialValues={config}>
+
+              {({ isSubmitting }) => {
+                return (
+                  <Form>
+                    <Modal.Header closeButton>
+                      <Modal.Title id="dialog_label">Configure Permissions</Modal.Title>
+                    </Modal.Header>
+
+                    <Modal.Body>
+                      <div>
+                        <Row>
+                          <Col sm={12}>
+                            <FormikInput type="checkbox"
+                                         name="allow_sharing_with_everyone"
+                                         id="shareWithEveryone"
+                                         label={(
+                                           <LabelSpan>Share with everyone</LabelSpan>
+                                         )} />
+                            <InputDescription help={<>Share with everyone option will be available</>} />
```

Comment:
This description is somehow confusing to me: the option will not be available if I disable the toggle. I suggest changing this to: "Control whether it is possible to share with everyone".

Reply:
By the way, would you mind changing the types for the `InputDescription` props from `ReactElement` to `ReactNode`? That will allow us to pass a `string` directly.